packages/widgetbook/lib/src/addons/device_addon/device_setting.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:widgetbook/src/addons/device_addon/frames/frames.dart';
+import 'package:widgetbook/widgetbook.dart';
+import 'package:widgetbook_addon/widgetbook_addon.dart';
+
+part 'device_setting.freezed.dart';
+
+@freezed
+class DeviceSetting extends WidgetbookAddOnModel with _$DeviceSetting {
+  @Assert('devices.isNotEmpty', 'devices cannot be empty')
+  factory DeviceSetting({
+    required List<Device?> devices,
+    required Device? activeDevice,
+    @Default(Orientation.portrait) Orientation orientation,
+    @Default(true) bool hasFrame,
+  }) = _DeviceSetting;
+
+  /// Sets the first [Device] within [devices] as the [activeDevice] on startup
+  factory DeviceSetting.firstAsSelected({
+    required List<Device?> devices,
+  }) {
+    return DeviceSetting(
+      devices: devices,
+      activeDevice: devices.first,
+    );
+  }
+
+  DeviceSetting._();
+
+  @override
+  Map<String, String> toQueryParameter() {
+    return {
+      'device': activeDevice?.name ?? 'none',
+      'orientation': orientation.name,
+      'frame': hasFrame.toString(),
+    };
+  }
+
+  @override
+  fromQueryParameter(Map<String, String> queryParameters) {
+    return queryParameters.containsKey('device') &&
+            queryParameters.containsKey('orientation') &&
+            queryParameters.containsKey('frame')
+        ? this.copyWith(
+            activeDevice: devices.firstWhere(
+              (device) => device?.name == queryParameters['device']!,
+              orElse: () => null,
+            ),
+            orientation: Orientation.values.byName(
+              queryParameters['orientation']!,
+            ),
+            hasFrame: queryParameters['frame'] == 'true',
+          )
+        : null;
+  }
+
+  FrameBuilder get frameBuilder {
+    if (activeDevice == null) {
+      return NoFrameBuilder(
+        setting: this,
+      );
+    }
+
+    if (!hasFrame) {
+      return WidgetbookFrameBuilder(
+        setting: this,
+      );
+    }
+
+    return DeviceFrameBuilder(
+      setting: this,
+    );
```

Comment:
How can we make the `device_frame` an optional dependency with this implementation?

Reply:
The ideal solution is extracting it to it's own package, for the simplicity of the public API of the `DeviceAddon`.
We can yet achieve this by exposing callbacks, but users will have to map Widgetbook's `Device` to `device_frame`'s `DeviceInfo` as well.

Reply:
Ideal solution sounds good. Do you think we can refactor it later?

Reply:
Yes, let's do it later with other addons if needed.